lib/platform/client/proxy_http.js

```diff
@@ -1,164 +1,228 @@
+                timeout: message.timeout || 0,
+                dataType: "text",
+                success: function(data, error, res) {
+                    var response = {
+                        statusCode: res.status,
+                        headers: getHeaders(res.getAllResponseHeaders())
+                    };
+
+                    complete_response = that._buildResponse(error, response, data);
+                    return Promise.resolve(complete_response);
+                },
+                error: function(res, data, error) {
+                    // Format abort response
+                    if(res.status === 600){
+                        data = JSON.parse(res.responseText).statusCode;
+                        let response = JSON.parse(res.responseText);
+                        complete_response = that._buildResponse("abort",response,{});
+                        return Promise.reject(complete_response);
+                    }
+                    var response = {
+                        statusCode: res.status,
+                        headers: getHeaders(res.getAllResponseHeaders())
+                    };
+
+                    if (data === "abort") {
+                        response.statusCode = "abort";
+                        res.responseText = "{}";
+                    }
+                    var json = res.responseText;
+
+                    complete_response = that._buildResponse(error, response, json);
+                    
+                    return Promise.reject(complete_response);
+                }
+            };
+            
+            return $.ajax(params).then((xhr)=>{
+                return complete_response;
+            }).catch((err)=>{
+                return complete_response;
+            });
+        },
+
+        makeRequestAsync: function(url, message) {
+            // Add our original destination to to headers,
+            // as some proxy implementations would rather
+            // use this.
+            message.headers["X-ProxyDestination"] = url;
+            message.headers["Splunk-Client"] = "splunk-sdk-javascript/" + SDK_VERSION;
+            if(message.response_timeout != undefined){
+                message.headers["Response-Timeout"] = message.response_timeout;
+            }
+
+            // Need to remove the hostname from the URL
+            var parsed = parseUri(url);
+            var prefixToRemove = "" + (parsed.protocol ? parsed.protocol : "") + "://" + parsed.authority;
+            url = url.replace(new RegExp(escape(prefixToRemove), "i"), "");
+            
+            // Now, we prepend the prefix
+            url = this.prefix + url;
+            
+            var that = this;
+            var complete_response;
+            var params = {
+                url: url,
+                type: message.method,
+                headers: message.headers,
+                data: message.body || "",
+                timeout: message.timeout || 0,

```

Comment:
I actually would suggest that we pick a reasonable timeout other than 0 but encourage users to set this themselves - I think needle and jquery probably handle this differently so it would be good to standardize on one, maybe 30 seconds? Leaving long running connections open on the splunkd side could definitely lead to performance issues - browsers will have different timeout defaults that also interact with this but I think consistency across clients would be best. I realize we haven't been doing this on the enterprise SDK side but the cloud SDKs took this approach. We should also call out the ability to set timeouts in our README as well as the default value that will be used if none is provided. What do you think?

Reply:
as discussed we'll create a follow up task for default timeout.